plugins/auth-backend/src/providers/google/provider.ts

```diff
@@ -53,16 +55,87 @@ export class GoogleAuthProvider
     res: express.Response,
     next: express.NextFunction,
   ) {
-    return passport.authenticate('google', (_, user) => {
-      postMessageResponse(res, {
+    return passport.authenticate('google', (err, user) => {
+      if (err) {
+        return postMessageResponse(res, {
+          type: 'auth-result',
+          error: new Error(`Google auth failed ${err.toString()}`),
+        });
+      }
+
+      const { refreshToken } = user;
+
+      if (!refreshToken) {
+        return postMessageResponse(res, {
+          type: 'auth-result',
+          error: new Error('Missing refresh token'),
+        });
+      }
+
+      delete user.refreshToken;
+
+      const options: CookieOptions = {
+        maxAge: THOUSAND_DAYS_MS,
+        secure: false,
```

Comment:
for now? eventually make configurable, and/or plain depend on NODE_ENV right?

Reply:
yes. this and domain will change based on NODE_ENV but the whole config and dev/prod setup would be a separate PR later.

Reply:
We need to support both environments simultaneously in the backend, it should be switched based on request parameters